presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PropertyDerivations.java

```diff
@@ -396,14 +396,14 @@ public ActualProperties visitJoin(JoinNode node, List<ActualProperties> inputPro
 
                     return ActualProperties.builderFrom(buildProperties.translate(column -> filterIfMissing(node.getOutputSymbols(), column)))
                             .local(ImmutableList.of())
-                            .unordered(unordered)
+                            .unordered(true)
```

Comment:
We should retain `probeProperties` like constants, ordered (unless spill enabled).

(both in RIGHT and FULL cases)